cluster/juju/layers/kubernetes-master/README.md

```diff
@@ -1,118 +1,102 @@
+The domain name to use for the Kubernetes cluster for kube-dns.
 
-To avoid data loss you must attach the storage before making the connection to
-the etcd cluster.
+#### enable-dashboard-addons
 
-## State Events
-While this charm is meant to be a top layer, it can be used to build other
-solutions.  This charm sets or removes states from the reactive framework that
-other layers could react appropriately. The states that other layers would be
-interested in are as follows:
+Enables the installation of Kubernetes dashboard, Heapster, Grafana, and
+InfluxDB.
 
-**kubelet.available** - The hyperkube container has been run with the kubelet
-service and configuration that started the apiserver, controller-manager and
-scheduler containers.
+# DNS for the cluster
 
-**proxy.available** - The hyperkube container has been run with the proxy
-service and configuration that handles Kubernetes networking.
+The DNS add-on allows the pods to have a DNS names in addition to IP addresses.
+The Kubernetes cluster DNS server (based off the SkyDNS library) supports 
+forward lookups (A records), service lookups (SRV records) and reverse IP 
+address lookups (PTR records). More information about the DNS can be obtained
+from the [Kubernetes DNS admin guide](http://kubernetes.io/docs/admin/dns/).
 
-**kubectl.package.created** - Indicates the availability of the `kubectl`
-application along with the configuration needed to contact the cluster
-securely. You will need to download the `/home/ubuntu/kubectl_package.tar.gz`
-from the kubernetes leader unit to your machine so you can control the cluster.
+# Adding optional storage
 
-**skydns.available** - Indicates when the Domain Name System (DNS) for the
-cluster is operational.
+The kubernetes-master charm handles different persistent storage devices that
+the cloud environments provide. See the 
+[Juju storage](https://jujucharms.com/docs/stable/charms-storage) documentation
+for more details about the supported clouds and storage pool types that are 
+available.
 
+Refer to the [Canonical Distribution of Kubernetes](https://jujucharms.com/canonical-kubernetes/) for more information
+about how add the Ceph charms that handle storage.
 
-# Kubernetes information
+# Additional Kubernetes information
 
  - [Kubernetes github project](https://github.com/kubernetes/kubernetes)
  - [Kubernetes issue tracker](https://github.com/kubernetes/kubernetes/issues)
- - [Kubernetes Documenation](http://kubernetes.io/docs/)
+ - [Kubernetes documentation](http://kubernetes.io/docs/)
  - [Kubernetes releases](https://github.com/kubernetes/kubernetes/releases)
 
 # Contact
 
- * Charm Author: Matthew Bruzek &lt;Matthew.Bruzek@canonical.com&gt;
- * Charm Contributor: Charles Butler &lt;Charles.Butler@canonical.com&gt;
+The kubernetes-master charm is free and open source operations software created
+by the containers team at Canonical. 
+
+Canonical also offers enterprise support and customisation services. Please
```

Comment:
I copied the en-uk way of spelling customisation from the page that was linked. Not sure what to do here, our README should agree with the marketing page.